app/models/invite.rb

```diff
@@ -160,6 +183,34 @@ def self.create_invite_by_email(email, invited_by, opts = nil)
     invite
   end
 
+  def self.generate_multiple_use_invite_link(invited_by:, max_redemptions_allowed: 5, expires_at: 1.month.from_now, group_ids: nil)
+    Invite.transaction do
+      create_args = {
+        invited_by: invited_by,
+        max_redemptions_allowed: max_redemptions_allowed.to_i,
+        expires_at: expires_at,
+        emailed_status: emailed_status_types[:not_required]
+      }
+      invite = Invite.create!(create_args)
+
+      if group_ids.present?
+        now = Time.zone.now
+        invited_groups = group_ids.map { |group_id| { group_id: group_id, invite_id: invite.id, created_at: now, updated_at: now } }
+        InvitedGroup.insert_all(invited_groups)
+      end
+
+      "#{Discourse.base_url}/invites/#{invite.invite_key}"
+    end
+  end
+
+  # redeem multiple use invite link
+  def redeem_invite_link(email: nil, username: nil, name: nil, password: nil, user_custom_fields: nil, ip_address: nil)
+    if is_invite_link? && !expired? && !redeemed? && !destroyed? && link_valid?
```

Comment:
@techAPJ I suggest we wrap this in a distributed mutex and within the mutex reload the invite object so that we get the latest `redemption_count`.

Reply:
Done.